cgi-bin/DW/Controller/Customize.pm

```diff
@@ -0,0 +1,654 @@
+
+    #handle post actions
+
+    if ( LJ::did_post() ) {
+        my @errors = LJ::Widget->handle_post( $POST,
+            qw(CustomizeTheme CustomTextModule MoodThemeChooser NavStripChooser S2PropGroup LinksList)
+        );
+        $ret .= LJ::bad_input(@errors) if @errors;
+    }
+
+    $vars->{content}     = $ret;
+    $vars->{layout_data} = get_layout_data();
+
+    $vars->{current_data} = get_current_data(1);
+
+    # Now we tell it what template to render and pass in our variables
+    return DW::Template->render_template( 'customize/index.tt', $vars );
+
+}
+
+## Functions to handle edits for both RPC and main endpoints
+
+sub set_theme {
+    my %opts = @_;
+    my $u    = LJ::get_effective_remote();
+    die "Invalid user." unless LJ::isu($u);
+
+    my $themeid  = $opts{apply_themeid} + 0;
+    my $layoutid = $opts{apply_layoutid} + 0;
+
+    my $theme;
+    if ($themeid) {
+        $theme = LJ::S2Theme->load_by_themeid( $themeid, $u );
+    }
+    elsif ($layoutid) {
+        $theme = LJ::S2Theme->load_custom_layoutid( $layoutid, $u );
+    }
+    else {
+        die "No theme id or layout id specified.";
+    }
+
+    LJ::Customize->apply_theme( $u, $theme ) or croak("Couldn't apply theme");
+    LJ::Hooks::run_hooks( 'apply_theme', $u );
+}
+
+sub set_layout {
+    my $post = shift;
+
+    my $u = LJ::get_effective_remote();
+    die "Invalid user." unless LJ::isu($u);
+
+    my %override;
+    my $layout_choice     = $post->{layout_choice};
+    my $layout_prop       = $post->{layout_prop};
+    my $show_sidebar_prop = $post->{show_sidebar_prop};
+    my $current_theme_id  = LJ::Customize->get_current_theme($u);
+    my %layouts           = $current_theme_id->layouts;
+
+    # show_sidebar prop is set to false/0 if the 1 column layout was chosen,
+    # otherwise it's set to true/1 and the layout prop is set appropriately.
+    if ( $show_sidebar_prop && $layout_choice eq "1" ) {
+        $override{$show_sidebar_prop} = 0;
+    }
+    else {
+        $override{$show_sidebar_prop} = 1 if $show_sidebar_prop;
+        $override{$layout_prop} = $layouts{$layout_choice} if $layout_prop;
+    }
+
+    my $style = LJ::S2::load_style( $u->prop('s2_style') );
+    die "Style not found." unless $style && $style->{userid} == $u->id;
+
+    LJ::Customize->save_s2_props( $u, $style, \%override );
+}
+
+sub set_journaltitles {
+    my $post = shift;
+
+    my $u = LJ::get_effective_remote();
+    die "Invalid user." unless LJ::isu($u);
+
+    my $eff_val = LJ::text_trim( $post->{title_value}, 0, LJ::std_max_length() );
+    $eff_val = "" unless $eff_val;
+    $u->set_prop( $post->{which_title}, $eff_val );
+}
+
+# Functions to render backend data in a format that is simpler to process in our templates/JS
+
+sub get_themechooser_data {
+    my $queryargs = shift;
+    my @themes;
+    my $u      = LJ::get_effective_remote();
+    my $remote = LJ::get_remote();
+    my %cats   = LJ::Customize->get_cats($u);
+
+    if ( $u ne $remote ) {
```

Comment:
Think we need to use same_user or compare userids or something.

Reply:
Other spots look like they compare $u->user and $remote->user so I updated it to that.